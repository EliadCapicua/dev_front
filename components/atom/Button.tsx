import { ActivityIndicator, StyleSheet, TouchableOpacity } from "react-native";
import { ThemedText } from "../ThemedText";

interface ButtonProps {
  text: string;
  onPress: () => void;
  isLoading?: boolean;
  isDisabled?: boolean;
  isDestructive?: boolean;
  isSecondary?: boolean;
}

export const Button = (props: ButtonProps) => {
  return (
    <TouchableOpacity
      style={[
        styles.container,
        props.isDisabled && { backgroundColor: "#E0E0E0" },
        props.isDestructive && { backgroundColor: "#FF0000" },
        props.isSecondary && { backgroundColor: "#DADAD7" },
      ]}
      onPress={props.onPress}
      disabled={props.isDisabled || props.isLoading}
    >
      {props.isLoading ? (
        <ActivityIndicator />
      ) : (
        <ThemedText
          type="default"
          style={[
            props.isDisabled && { opacity: 0.5 },
            props.isDestructive && { color: "#FFFFFF" },
            props.isSecondary && { color: "#000000" },
          ]}
        >
          {props.text}
        </ThemedText>
      )}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    width: "100%",
    backgroundColor: "#F8E930",
    padding: 10,
    borderRadius: 5,
  },
});
