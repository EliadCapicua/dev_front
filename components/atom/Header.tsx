import { SafeAreaView, StyleSheet, Platform, View } from "react-native";
import { ThemedText } from "../ThemedText";
import { FontAwesome6 } from "@expo/vector-icons";

interface HeaderProps {
  title: string;
  onLeftPress?: () => void;
}

export const Header = (props: HeaderProps) => {
  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.headerRow}>
        {props.onLeftPress && (
          <FontAwesome6
            name="arrow-left"
            size={24}
            color="#0777E0"
            style={styles.leftIcon}
            onPress={props.onLeftPress}
          />
        )}
        <View
          style={[
            styles.headerTitle,
            { marginRight: props.onLeftPress ? 44 : 0 },
          ]}
        >
          <FontAwesome6
            name="money-bills"
            size={24}
            color="#0777E0"
            style={styles.icon}
          />
          <ThemedText type="title" style={styles.title}>
            {props.title}
          </ThemedText>
        </View>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    display: "flex",
    alignItems: "center",
    width: "100%",
    marginTop: Platform.OS === "ios" ? 44 : 0,
    borderBottomWidth: 1,
    borderBottomColor: "#E0E0E0",
    height: 44,
  },
  headerRow: {
    display: "flex",
    width: "100%",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
  },
  headerTitle: {
    flex: 1,
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
  },
  leftIcon: {
    marginLeft: 20,
  },
  icon: {
    marginBottom: 4,
  },

  title: {
    textAlign: "center",
    marginLeft: 8,
  },
});
