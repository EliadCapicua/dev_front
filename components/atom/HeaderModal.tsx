import { Ionicons } from "@expo/vector-icons";
import { TouchableOpacity, View, StyleSheet } from "react-native";

interface HeaderModalProps {
  onPress: () => void;
}

export const HeaderModal = (props: HeaderModalProps) => {
  return (
    <View style={styles.header}>
      <TouchableOpacity onPress={props.onPress}>
        <Ionicons name="close" size={24} color="black" />
      </TouchableOpacity>
    </View>
  );
};
const styles = StyleSheet.create({
  header: {
    display: "flex",
    width: "100%",
    flexDirection: "row",
    alignItems: "flex-end",
    justifyContent: "flex-end",
    paddingHorizontal: 8,
    paddingVertical: 6,
    borderBottomWidth: 1,
    borderBottomColor: "lightgray",
  },
});
