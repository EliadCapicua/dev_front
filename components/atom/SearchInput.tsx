import { TextInput, View, StyleSheet } from "react-native";

export interface SearchInputProps {
  placeholder: string;
}

export const SearchInput = (props: SearchInputProps) => {
  return (
    <View style={styles.container}>
      <TextInput placeholder={props.placeholder} style={styles.input} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    display: "flex",
    width: "100%",
    flexDirection: "row",
    alignItems: "flex-start",
    justifyContent: "flex-start",
    backgroundColor: "#FFFF",
    borderRadius: 8,
    borderWidth: 1,
    borderColor: "#E0E0E0",
    padding: 8,
  },
  input: {
    flex: 1,
    padding: 4,
    fontSize: 16,
    fontWeight: "600",
    color: "#333",
  },
});
