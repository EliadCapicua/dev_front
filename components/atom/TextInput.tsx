import { View, StyleSheet, TextInput } from "react-native";
import { ThemedText } from "../ThemedText";
import { useEffect, useState } from "react";

export interface TextComponentProps {
  label?: string;
  placeholder: string;
  onChangeText: (text: string) => void;
  value: string;
  isValid?: boolean;
  error?: string;
  isOptional?: boolean;
}

export const TextComponent = (props: TextComponentProps) => {
  const [value, onChangeText] = useState("");

  const handleOnChangeText = (text: string) => {
    onChangeText(text);
    props.onChangeText(text);
  };

  useEffect(() => {
    onChangeText(props.value);
    return () => {};
  }, [props.value]);

  return (
    <View style={styles.container}>
      <ThemedText style={styles.label}>{props.label}</ThemedText>
      <TextInput
        placeholder={props.placeholder}
        style={[
          styles.input,
          !props.isValid && !props.isOptional && { borderColor: "#FF0000" },
        ]}
        onChangeText={handleOnChangeText}
        value={value}
        autoCapitalize="none"
      />
      {!props.isValid && !props.isOptional && (
        <ThemedText style={styles.error}>
          {props.error || "Este campo es requerido"}
        </ThemedText>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    display: "flex",
    width: "100%",
    flexDirection: "column",
    alignItems: "flex-start",
    justifyContent: "flex-start",
    marginVertical: 8,
  },
  label: {
    fontSize: 16,
    fontWeight: "600",
    color: "#333",
    marginBottom: 4,
  },
  input: {
    height: 45,
    padding: 8,
    fontSize: 16,
    fontWeight: "600",
    color: "#333",
    width: "100%",
    borderWidth: 1,
    borderColor: "#E0E0E0",
    borderRadius: 8,
    textTransform: "lowercase",
  },
  error: {
    fontSize: 14,
    color: "#FF0000",
  },
});
