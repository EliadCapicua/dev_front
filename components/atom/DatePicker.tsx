import {
  View,
  TouchableOpacity,
  Modal,
  Platform,
  StyleSheet,
} from "react-native";
import React, { useState } from "react";
import DateTimePicker, {
  DateTimePickerEvent,
  DateTimePickerAndroid,
} from "@react-native-community/datetimepicker";
import { ThemedText } from "../ThemedText";
import { Button } from "./Button";
import { formatDate } from "@/utils/date";

interface DatePickerProps {
  label?: string;
  value: string;
  onChange?: (value: string) => void;
  placeholder: string;
  isOptional?: boolean;
  minDate?: Date;
  maxDate?: Date;
  isValid?: boolean;
  error?: string;
}

export const DatePicker: React.FC<DatePickerProps> = ({
  label,
  value,
  onChange,
  placeholder,
  minDate,
  maxDate,
}) => {
  const [date, setDate] = useState(value ? new Date(value) : undefined);
  const [dateString, setDateString] = useState(
    value ? new Date(value).toDateString() : ""
  );
  const [isModalOpen, setIsModalOpen] = useState(false);

  const onChangeDate = (
    event: DateTimePickerEvent,
    selectedDate: Date | undefined
  ) => {
    const currentDate = selectedDate || date;
    if (!currentDate) return;
    setDate(currentDate);
    setDateString(formatDate(currentDate.toDateString()));
    onChange?.(currentDate?.toISOString() || "");
  };

  const onPress = () => {
    if (Platform.OS === "ios") {
      setIsModalOpen(true);
    } else {
      DateTimePickerAndroid.open({
        value: date || new Date(),
        onChange: onChangeDate,
        mode: "date",
        maximumDate: maxDate || new Date(),
        minimumDate: minDate || new Date(1920, 0, 1),
      });
    }
  };

  return (
    <View style={styles.container}>
      <ThemedText style={styles.label}>{label}</ThemedText>
      <TouchableOpacity onPress={onPress} style={styles.input}>
        <ThemedText style={[styles.date, date ? {} : { color: "#E0E0E0" }]}>
          {date ? dateString : placeholder}
        </ThemedText>
      </TouchableOpacity>

      {Platform.OS === "ios" && (
        <Modal animationType="fade" transparent={true} visible={isModalOpen}>
          <View
            style={[styles.modal, { backgroundColor: "rgba(0, 0, 0, 0.5)" }]}
          >
            <View style={styles.modalContent}>
              <DateTimePicker
                value={date || new Date()}
                mode="date"
                display="spinner"
                onChange={onChangeDate}
                maximumDate={maxDate || new Date()}
                minimumDate={minDate || new Date(1920, 0, 1)}
              />
              <View style={{ flexDirection: "row", justifyContent: "center" }}>
                <Button text="Select" onPress={() => setIsModalOpen(false)} />
              </View>
            </View>
          </View>
        </Modal>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    display: "flex",
    width: "100%",
    flexDirection: "column",
    alignItems: "flex-start",
    justifyContent: "flex-start",
    marginVertical: 8,
  },
  modal: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    height: "100%",
  },
  modalContent: {
    backgroundColor: "white",
    padding: 16,
    borderRadius: 8,
    width: "80%",
  },
  label: {
    fontSize: 16,
    fontWeight: "600",
    color: "#333",
    marginBottom: 4,
  },
  date: {
    fontSize: 16,
    fontWeight: "600",
    color: "#333",
  },
  input: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    height: 44,
    padding: 4,
    fontSize: 16,
    fontWeight: "600",
    color: "#333",
    width: "100%",
    borderWidth: 1,
    borderColor: "#E0E0E0",
    borderRadius: 8,
  },
});
