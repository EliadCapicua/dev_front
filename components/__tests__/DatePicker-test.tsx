import * as React from "react";
import renderer from "react-test-renderer";
import { DatePicker } from "../atom/DatePicker";

it(`renders correctly`, () => {
  const tree = renderer
    .create(<DatePicker placeholder="Snapshot test!" value="" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
