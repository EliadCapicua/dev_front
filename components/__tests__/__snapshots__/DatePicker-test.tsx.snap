// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "marginVertical": 8,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "color": "#333",
          "fontSize": 16,
          "fontWeight": "600",
          "marginBottom": 4,
        },
      ]
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "#E0E0E0",
        "borderRadius": 8,
        "borderWidth": 1,
        "color": "#333",
        "display": "flex",
        "flexDirection": "row",
        "fontSize": 16,
        "fontWeight": "600",
        "height": 44,
        "justifyContent": "space-between",
        "opacity": 1,
        "padding": 4,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          [
            {
              "color": "#333",
              "fontSize": 16,
              "fontWeight": "600",
            },
            {
              "color": "#E0E0E0",
            },
          ],
        ]
      }
    >
      Snapshot test!
    </Text>
  </View>
</View>
`;
